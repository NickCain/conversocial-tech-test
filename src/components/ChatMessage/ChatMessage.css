.Chat-message {
    width: 100%;
    height: 40px;
}

.Chat-message.user {
    text-align: left;
}

.Chat-message.user .message {
    float: right;
    background: lightblue;
    width: 250px;
    padding: 20px;
}

.Chat-message.app {
    text-align: right;
}

.Chat-message.app .message {
    background: lightcoral;
    width: 250px;
    padding: 20px;
}
