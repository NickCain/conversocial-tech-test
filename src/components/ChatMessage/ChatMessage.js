import React, { Component } from 'react';

import './ChatMessage.css';

class ChatMessage extends Component {
  render() {
    const { message } = this.props;
    return (
      <div className="Chat-message">
        {message}
      </div>
    );
  }
}

export default ChatMessage;
