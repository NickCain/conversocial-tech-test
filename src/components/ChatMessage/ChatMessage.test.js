import { mount, shallow } from 'enzyme';
import React from 'react';
import ChatMessage from './ChatMessage.js'

let wrapper;
let shallowWrapper;

describe('ChatMessage - component', () => {
  describe('ChatMessage renders', () => {
    beforeEach(() => {
      wrapper = mount(
        <ChatMessage />
      );
      shallowWrapper = shallow(
        <ChatMessage />
      );
    });

    it('should render the ChatMessage component', () => {
      expect(shallowWrapper).toMatchSnapshot();
      expect(wrapper.length).toBeGreaterThan(0);
    });
  });
});
