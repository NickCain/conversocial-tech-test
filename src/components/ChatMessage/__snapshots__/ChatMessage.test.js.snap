// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessage - component ChatMessage renders should render the ChatMessage component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="Chat-message"
/>,
  "nodes": Array [
    <div
      className="Chat-message"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ChatMessage />,
      "_debugID": 4,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ChatMessage {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="Chat-message"
/>,
        "_debugID": 5,
        "_renderedOutput": <div
          className="Chat-message"
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ChatMessage />,
}
`;
